### PR TITLE
[server] Do not open storage engines for invalid store versions

### PR DESCRIPTION
**Pull request by majisourav99**
## Summary, imperative, start upper case, don't end with a period

Do not open storage engines for invalid store versions. Often times due to failed pushes/stuck ingestion tasks or some unknown reasons the server host rocksDB path would contain many old store version folders. During restart of the server currently we try to open each and every one of them, which is a waste of resources. This PR validate the store versions before opening them.

## How was this PR tested?

CI build
## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java [hunk 1]
```diff
@@ -172,6 +172,17 @@ private void restoreAllStores(
         AbstractStorageEngine storageEngine;
 
         try {
+          String store = Version.parseStoreFromVersionTopic(storeName);
```
Comment (adamxchen) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java:175]: Can we move these lines up to line 168 so we even don't have to load the config for it?

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java [hunk 2]
```diff
@@ -172,6 +172,17 @@ private void restoreAllStores(
...
+          int versionNum = Version.parseVersionFromKafkaTopicName(storeName);
+
+          Optional<Version> version = storeRepository.getStoreOrThrow(store).getVersion(versionNum);
```
[the remark this answers is not in the document]
Comment (sixpluszero) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java:178]: Maybe we should do the logging and skip as you do below for its version

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java (hunk 1 above)
Comment (majisourav99) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java:175]: I didnt do that as wanted a smaller try block.  But sure.

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (majisourav99) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java:178]: yes, we will catch and skip it.

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java [hunk 3]
```diff
@@ -165,6 +166,24 @@ private void restoreAllStores(
         /**
          * Setup store-level persistence type based on current database setup.
          */
+
+        if (!VeniceSystemStoreUtils.isSystemStore(storeName)) {
```
Comment (gaojieliu) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java:170]: Why don't we check system store here?

**Review by gaojieliu — CHANGES_REQUESTED** [review]

**Pull request closed without merging**